### ISSUE TITLE
Bees 3.0 runtimes

### ISSUE THREAD

**Issue by DamianX** [d3athrow/vgstation13#17387]
```
x6131 bees_mob.dm,273: Cannot read .len
  proc name: Life (/mob/living/simple_animal/bee/Life)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (254,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```
x172 bees_mob.dm,422: Cannot read .len
  proc name: Life (/mob/living/simple_animal/bee/Life)
  src: the swarm of angry bees (/mob/living/simple_animal/bee)
  src.loc: the floor (254,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of angry bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```
x156 bees_mob.dm,246: Cannot read null.angery
  proc name: panic attack (/mob/living/simple_animal/bee/proc/panic_attack)
  usr: Ola Stanford (jubisloviu) (/mob/living/carbon/human)
  usr.loc: The floor (258, 237, 1) (/turf/simulated/floor)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (255,238,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): panic attack(Ola Stanford (/mob/living/carbon/human))
  the swarm of bees (/mob/living/simple_animal/bee): bullet act(the fire plume (/obj/item/projectile/bullet/fire_plume), "chest")
  the fire plume (/obj/item/projectile/bullet/fire_plume): to bump(the swarm of bees (/mob/living/simple_animal/bee), 1)
  the floor (255,238,1) (/turf/simulated/floor): Enter(the fire plume (/obj/item/projectile/bullet/fire_plume), the floor (256,238,1) (/turf/simulated/floor))
  the fire plume (/obj/item/projectile/bullet/fire_plume): Move(the floor (255,238,1) (/turf/simulated/floor), 0, 0, 0, 0)
  the fire plume (/obj/item/projectile/bullet/fire_plume): make bresenham step(6, 3, 8, 1)
  the fire plume (/obj/item/projectile/bullet/fire_plume): bresenham step(6, 3, 8, 1)
  the fire plume (/obj/item/projectile/bullet/fire_plume): process step()
  the fire plume (/obj/item/projectile/bullet/fire_plume): process step()
  the fire plume (/obj/item/projectile/bullet/fire_plume): process()
```

```
x155 bees_mob.dm,198: Cannot read .len
  proc name: adjustBruteLoss (/mob/living/simple_animal/bee/adjustBruteLoss)
  usr: Phallic Member (adamhunt9) (/mob/living/carbon/human)
  usr.loc: The floor (253, 238, 1) (/turf/simulated/floor)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (253,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): adjustBruteLoss(5)
  the swarm of bees (/mob/living/simple_animal/bee): apply damage(5, "brute", null, 0, null, 0)
  Phallic Member (/mob/living/carbon/human): unarmed attack mob(the swarm of bees (/mob/living/simple_animal/bee))
  the swarm of bees (/mob/living/simple_animal/bee): attack hand(Phallic Member (/mob/living/carbon/human), "icon-x=36;icon-y=8;left=1", 1)
  Phallic Member (/mob/living/carbon/human): UnarmedAttack(the swarm of bees (/mob/living/simple_animal/bee), 1, "icon-x=36;icon-y=8;left=1")
  Phallic Member (/mob/living/carbon/human): ClickOn(the swarm of bees (/mob/living/simple_animal/bee), "icon-x=36;icon-y=8;left=1")
  the swarm of bees (/mob/living/simple_animal/bee): Click("floor", "infowindow.info", "icon-x=36;icon-y=8;left=1")
```

```
x131 bees_mob.dm,395: bad index
  proc name: Life (/mob/living/simple_animal/bee/Life)
  src: the swarm of angry bees (/mob/living/simple_animal/bee)
  src.loc: the floor (276,250,1) (/turf/simulated/floor)
  call stack:
  the swarm of angry bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```x28 bees_mob.dm,215: Cannot execute .Add().
  proc name: addBee (/mob/living/simple_animal/bee/proc/addBee)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (254,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): addBee(/datum/bee (/datum/bee))
  the swarm of angry bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```
x28 bees_mob.dm,581: Cannot read .len
  proc name: update icon (/mob/living/simple_animal/bee/update_icon)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (254,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): update icon()
  the swarm of angry bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```
x19 bees_mob.dm,262: Cannot execute .Find().
  proc name: add plants (/mob/living/simple_animal/bee/proc/add_plants)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (254,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): add plants(/list (/list))
  the swarm of angry bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```
x15 bees_mob.dm,129: Cannot read .len
  proc name: Cross (/mob/living/simple_animal/bee/Cross)
  usr: Ola Stanford (jubisloviu) (/mob/living/carbon/human)
  usr.loc: The floor (258, 237, 1) (/turf/simulated/floor)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (255,238,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): Cross(the fire plume (/obj/item/projectile/bullet/fire_plume), the floor (256,238,1) (/turf/simulated/floor), 1.5, 0)
  the floor (255,238,1) (/turf/simulated/floor): Enter(the fire plume (/obj/item/projectile/bullet/fire_plume), the floor (256,238,1) (/turf/simulated/floor))
  the fire plume (/obj/item/projectile/bullet/fire_plume): Move(the floor (255,238,1) (/turf/simulated/floor), 0, 0, 0, 0)
  the fire plume (/obj/item/projectile/bullet/fire_plume): make bresenham step(6, 3, 8, 1)
  the fire plume (/obj/item/projectile/bullet/fire_plume): bresenham step(6, 3, 8, 1)
  the fire plume (/obj/item/projectile/bullet/fire_plume): process step()
  the fire plume (/obj/item/projectile/bullet/fire_plume): process step()
  the fire plume (/obj/item/projectile/bullet/fire_plume): process()
```

```
x11 bees_items.dm,54: Cannot read null.common_name
  proc name: afterattack (/obj/item/weapon/bee_net/afterattack)
  usr: Doctor Frank (vampirebat74) (/mob/living/carbon/human)
  usr.loc: The floor (253, 238, 1) (/turf/simulated/floor)
  src: the bug net (/obj/item/weapon/bee_net)
  src.loc: Doctor Frank (/mob/living/carbon/human)
  call stack:
  the bug net (/obj/item/weapon/bee_net): afterattack(the floor (253,239,1) (/turf/simulated/floor), Doctor Frank (/mob/living/carbon/human), 1, "icon-x=25;icon-y=3;left=1;scre...")
  Doctor Frank (/mob/living/carbon/human): ClickOn(the floor (253,239,1) (/turf/simulated/floor), "icon-x=25;icon-y=3;left=1;scre...")
  the floor (253,239,1) (/turf/simulated/floor): Click(the floor (253,239,1) (/turf/simulated/floor), "mapwindow.map", "icon-x=25;icon-y=3;left=1;scre...")
```

```
x5 bees_mob.dm,207: Cannot read .len
  proc name: adjustBruteLoss (/mob/living/simple_animal/bee/adjustBruteLoss)
  usr: Ola Stanford (jubisloviu) (/mob/living/carbon/human)
  usr.loc: The floor (258, 237, 1) (/turf/simulated/floor)
  src: the swarm of bees (/mob/living/simple_animal/bee)
  src.loc: the floor (255,238,1) (/turf/simulated/floor)
  call stack:
  the swarm of bees (/mob/living/simple_animal/bee): adjustBruteLoss(0)
  the swarm of bees (/mob/living/simple_animal/bee): bullet act(the fire plume (/obj/item/projectile/bullet/fire_plume), "chest")
  the swarm of bees (/mob/living/simple_animal/bee): bullet act(the fire plume (/obj/item/projectile/bullet/fire_plume), "chest")
  the fire plume (/obj/item/projectile/bullet/fire_plume): to bump(the swarm of bees (/mob/living/simple_animal/bee), 1)
  the floor (255,238,1) (/turf/simulated/floor): Enter(the fire plume (/obj/item/projectile/bullet/fire_plume), the floor (256,238,1) (/turf/simulated/floor))
  the fire plume (/obj/item/projectile/bullet/fire_plume): Move(the floor (255,238,1) (/turf/simulated/floor), 0, 0, 0, 0)
  the fire plume (/obj/item/projectile/bullet/fire_plume): make bresenham step(6, 3, 8, 1)
  the fire plume (/obj/item/projectile/bullet/fire_plume): bresenham step(6, 3, 8, 1)
  the fire plume (/obj/item/projectile/bullet/fire_plume): process step()
  the fire plume (/obj/item/projectile/bullet/fire_plume): process step()
  the fire plume (/obj/item/projectile/bullet/fire_plume): process()
```

```
x2 bees_mob.dm,354: Cannot execute null.has reagent().
  proc name: Life (/mob/living/simple_animal/bee/Life)
  src: the swarm of angry bees (/mob/living/simple_animal/bee)
  src.loc: the floor (230,239,1) (/turf/simulated/floor)
  call stack:
  the swarm of angry bees (/mob/living/simple_animal/bee): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

```
x1 bees_apiary.dm,337: Cannot read null.angery
  proc name: angry swarm (/obj/machinery/apiary/proc/angry_swarm)
  usr: Doctor Frank (vampirebat74) (/mob/living/carbon/human)
  usr.loc: Carpet (283, 252, 1) (/turf/simulated/floor/carpet)
  src: the apiary tray (/obj/machinery/apiary)
  src.loc: the floor (284,252,1) (/turf/simulated/floor/wood)
  call stack:
  the apiary tray (/obj/machinery/apiary): angry swarm(Doctor Frank (/mob/living/carbon/human))
  the apiary tray (/obj/machinery/apiary): attackby(the empty packet of BeezEez (/obj/item/trash/beezeez), Doctor Frank (/mob/living/carbon/human), "icon-x=20;icon-y=9;left=1;scre...")
  Doctor Frank (/mob/living/carbon/human): ClickOn(the apiary tray (/obj/machinery/apiary), "icon-x=20;icon-y=9;left=1;scre...")
  the apiary tray (/obj/machinery/apiary): Click(the floor (284,252,1) (/turf/simulated/floor/wood), "mapwindow.map", "icon-x=20;icon-y=9;left=1;scre...")
```

Holy shit.
Note that each of these quote blocks is a separate runtime, no duplicates.
@DeityLink

**Comment by DeityLink** [comment]
geez that's a lot of runtimes. I'll get to them later this week end.

**Comment by gbasood** [comment]
Alright so I don't personally see a way bees could be anything but a list, so that's the majority of those runtimes that I don't see why they'd ever happen unless something else was setting the bee's var
It isn't null because that'd be `null.len` instead, not `.len`

Also, I did a test, and casting an obj or datum or number to a list does not reproduce that specific `cannot read .len` error.
It's likely we just ran out of memory yet again

**Comment by DeityLink** [comment]
How many bees were there exactly? maybe we've finally reached a cap

**Comment by DamianX** [comment]
Should be fixed by #18448, if they keep happening open a new issue with the correct new information.
